### ISSUE TITLE
Comment and Close Notes

### ISSUE THREAD

**Issue by GH0STsama** [MarcusWolschon/osmeditor4android#1654]
A `comment and close` (the notes) option like the one that **iD** brings would be quite useful, I've searched but I can't find any way to do it, I assume that the app doesn't have this option?
 If so, how about the idea of ​​adding it?

**Comment by simonpoole** [comment]
Well you just need to set the status to closed, write the comment (or the other way around) then save or directly upload. Not convinced that there is enough savings to justify an additional button.

**Comment by legzdev** [comment]
yes, but that way then first comment, and then close the note, when you do it from **iD** you can close the note directly with the comment, and not as two different actions 🤔

**Comment by simonpoole** [comment]
Not sure I follow, you really only have to set the spinner to "Closed", compared to the typing of the comment that is a split-second thing.

Adding a button to do "Close and Save" implies changing the Task UI from a standard Alert dialog to a custom dialog with 4 instead of 3 buttons and the associated layout issues. There is just barely space for three buttons as is (not even considering translations).

**Comment by GH0STsama** [comment]
> Not sure I follow, you really only have to set the spinner to "Closed", compared to the typing of the comment that is a split-second thing.
> 
> Adding a button to do "Close and Save" implies changing the Task UI from a standard Alert dialog to a custom dialog with 4 instead of 3 buttons and the associated layout issues. There is just barely space for three buttons as is (not even considering translations).

you are right, it has quite a few things, one more button would make a total imbalance..

**Comment by GHOSTsama2503** [comment]
Still, this would be much more comfortable when working with a large number of notes with tools such as: https://resultmaps.neis-one.org/osm-notes.

**Comment by GHOSTsama2503** [comment]
@simonpoole what if instead of changing the interface, you simply change the behavior of the buttons?

If the spinner is on "Closed" and there is a comment, instead of commenting and then closing, close directly with the comment.
